### ISSUE TITLE
Generate TWAs that can take Digital Goods payments

### ISSUE THREAD

**Issue by PEConn** [GoogleChromeLabs/bubblewrap#376]
The Digital Goods API in Android Browser Helper is not yet 100% complete, but is a good part of the way there. We should start looking into making bubblewrap generate TWAs that can use it.

**Comment by zorfling** [comment]
Is there any update on this @PEConn ?

**Comment by PEConn** [comment]
Heya,

The basics are in place, and bubblewrap should be able to generate TWAs that support Play Billing since #403. We still have to make this nicer for users to use (eg, prompting for it in the interactive prompt), but if you generate a TWA, modify the `twa-manifest.json` as follows and update and build, you should get a Play Billing supporting TWA.

You've got to have in your twa-manifest.json:
```
  ...,
  "enableNotifications": true,
  "features": {
    "playBilling": {
      "enabled": true
    }   
  },  
  "alphaDependencies": {
    "enabled": true
  },
```

**Comment by PEConn** [comment]
This should be published in v1.8.0

**Comment by csbenjamin** [comment]
Hi @PEConn
How can I request user payment (subscription) from javascript once I enable this feature in twa-manifest.json? Sorry if this is a dumb question, I am new to android world. Is there somewhere (documentation) I can go for such info?

**Comment by PEConn** [comment]
Heya,

Our documentation is still a little in progress at the moment, [this][1] is what we have at the moment. The good news is that you shouldn't need to touch anything in the Android world, just the web world.

So for a brief explanation, Play Billing with TWAs is split up into two parts:
* The Digital Goods API, which allows you to check the prices and details (this is important because you may have different prices in different countries and the Play Store should be the source of truth), acknowledge/consume purchases and list what the user has already purchased.
* The Payment Request API, which allows you to actually make purchases.

The Payment Request API has been around for a while, but has had one or two changes to make it work with TWA Play Billing, the Digital Goods API is a brand new thing (in Chrome 88). Additionally, you'll need to apply to the [origin trial][2] to get it working on your website.

If you visit https://beer.conn.dev and go to the bottom, you can download my test app that uses Digital Goods (if you complete the Payment Flow for anything under "Production Goods" you will be charged actual money, so be careful), however if you're struggling with getting things working, you can have a look at the source for it:

* https://beer.conn.dev/index.html (View source)
* https://beer.conn.dev/js/payments.js

(Sorry for the quality of the code - I'm the opposite of you, I know more Android than I do web dev.)

Hope that helps! We've only just launched and there will be better docs coming soon.

[1]: https://developers.google.com/web/android/trusted-web-activity/whats-new#google_play_billing
[2]: https://developer.chrome.com/origintrials/#/register_trial/-5451607348931985407

**Comment by dseeker** [comment]
Very exciting, I experimented deploying a PWA site in Bubblewrap and added the PaymentRequest code and it was very easy for make the purchase prompt appear. To acknowledge the purchase I was able to send the token via API to a node.js backend where the token is verified, added to a db and acknowledged. this is a game changer!

I'm now struggling to understand how to make the purchase consumed to the user can purchase again the same inapp product. I looked around for a server-side solution for it but as I understand the consume API does not exist in the play store node library, it can only be done in the client, is this correct?

I see in your code, @PEConn, how to acknowledge via the DigitalGoods api, the difference between them is just passing the string type 'onetime' vs 'repeatable'
```
  service = await window.getDigitalGoodsService(PAYMENT_METHOD);
  await service.acknowledge(token, type);
```

since this API is so new, regarding deployment, how to make sure that all your app users are able to properly purchase if they don't have a chrome version that supports the Digital Goods API? is there a way to embed the working chrome version in the application with bubblewrap, or somehow prompt an update of their chrome version before starting the app?

I'm also curious, if that's not possible, would there be a way to manually trigger a new activity that's added to the bubblewrap generated manifest to consume the product on the client?
